crates/uv/tests/it/python_find.rs

```diff
@@ -663,3 +664,82 @@ fn python_find_venv_invalid() {
     ----- stderr -----
     "###);
 }
+
+/// See: <https://github.com/astral-sh/uv/issues/11825>
+#[test]
+#[cfg(unix)]
+fn python_required_python_major_minor() {
+    let context: TestContext = TestContext::new_with_versions(&["3.11", "3.12"]);
+
+    // Find the Python 3.11 executable.
+    let path =
+        String::from_utf8(context.python_find().arg("3.11").output().unwrap().stdout).unwrap();
+    let path = Path::new(path.trim());
+
+    // Symlink it to `python3.11`.
+    fs_err::create_dir_all(context.temp_dir.child("child")).unwrap();
+    std::os::unix::fs::symlink(path, context.temp_dir.child("child").join("python3.11")).unwrap();
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml
+        .write_str(indoc! {r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.11.4, <3.12"
```

Comment:
As a note, you can just use this in `python find` directly instead of using a `pyproject.toml` to avoid indirection. I don't have strong feelings otherwise.